src/Sylius/Behat/Context/Api/Shop/CartContext.php

```diff
@@ -794,17 +794,40 @@ private function hasItemWithNameAndQuantity(Response $response, string $productN
         return false;
     }
 
-    private function checkProductQuantity(
+    private function checkProductQuantityByAdmin(
         Response $cartResponse,
         string $productName,
         int $quantity
     ): void {
+        $this->checkProductQuantity($cartResponse, $productName, $quantity, 'admin');
+    }
+
+    private function checkProductQuantityByCustomer(
+        Response $cartResponse,
+        string $productName,
+        int $quantity,
+    ): void {
+        $this->checkProductQuantity($cartResponse, $productName, $quantity, 'customer');
+    }
+
+    private function checkProductQuantity(
```

Comment:
Personally, I'm not a fan of this solution. In order to reduce duplication, we've introduced 4 if statements

Reply:
Honestly, I don't understand what exactly you mean. Could you describe in other words what I should change?